MovingBlocks/Terasology
# test for telemetry

### Contains

This PR is for the test, please(of course) **don't** merge it

The first metric sent to the [telemetry server](https://github.com/GabrielXia/telemetry). Informations collected are os, processors, etc. Kibana in the server(now it's localhost) should look like this:

![telemetry](https://cloud.githubusercontent.com/assets/17897358/26754861/da8ad764-4882-11e7-8b4c-21b1de013cde.png)


### How to test

1. Firstly exclude jackson-core in CR libs: [CR#39](https://github.com/MovingBlocks/CrashReporter/pull/39). `gradlew fetchLibsCrashReporter` if you didn't fetch CR libs
2. `gradlew idea` to load snowplow-track lib
3. You might need to setup server locally: https://github.com/GabrielXia/telemetry#setup
4. `docker-compose up` to start all the server(it might take a while to build and start). Then start the game, terasology will send information when game `initialize()`
5. Go to kibana page http://localhost:5601:
- Firstly setting index pattern, go to `setting` page. In Index name or pattern, enter `good` then in Time-field name choose `collector_tstamp` to visualise good metric data
- Then normally you can see the system context in `Discover page`

please let me now when there is an issue :)

### Outstanding before merging

- [x] Discuss about when to send the event. Maybe certain minutes after the information is got by the game. That will give users time to choose whether they want the information to be send or not
